File: VB/TextEditPasswordStrengthBar/TextEditPwdStrength/TextEditPwdStrengthViewInfo.vb
Imports System.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Repository

Namespace TextEditPasswordStrengthBar

    Friend Class TextEditPwdStrengthViewInfo
        Inherits TextEditViewInfo

        Public Sub New(ByVal item As RepositoryItem)
            MyBase.New(item)
        End Sub

        Public ReadOnly Property RepositoryItem As RepositoryItemTextEditPwdStrength
            Get
                Return TryCast(Item, RepositoryItemTextEditPwdStrength)
            End Get
        End Property

        Protected Overrides Function CalcMaskBoxRect(ByVal content As System.Drawing.Rectangle, ByRef contextImageBounds As Rectangle) As System.Drawing.Rectangle
            Dim r As Rectangle = MyBase.CalcMaskBoxRect(content, contextImageBounds)
            r.Height -= 1
            r.Y = 1
            Return r
        End Function
    End Class
End Namespace

File: CS/TextEditPasswordStrengthBar/TextEditPwdStrength/TextEditPwdStrengthViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Repository;

namespace TextEditPasswordStrengthBar
{
    class TextEditPwdStrengthViewInfo : TextEditViewInfo
    {
        public TextEditPwdStrengthViewInfo(RepositoryItem item) : base(item) { }

        public RepositoryItemTextEditPwdStrength RepositoryItem
        {
            get { return this.Item as RepositoryItemTextEditPwdStrength; }
        }

        protected override System.Drawing.Rectangle CalcMaskBoxRect(System.Drawing.Rectangle content, ref Rectangle contextImageBounds)
        {
            Rectangle r = base.CalcMaskBoxRect(content, ref contextImageBounds);
            r.Height -= 1;
            r.Y = 1;
            return r;
        }
    }
}
